affinidi-messaging-mediator/src/messages/protocols/message_pickup.rs

```diff
@@ -331,6 +331,26 @@ pub(crate) async fn delivery_request(
                 }
             };
 
+        if session.did != recipient_did {
+            return Err(MediatorError::Unauthorized(
+                session.session_id.clone(),
+                format!(
+                    "Session DID \"{}\" doesn't match recipient DID \"{}\"",
+                    session.did, recipient_did
+                ),
+            ));
+        }
+
+        if limit < 1 || limit > 100 {
```

Comment:
Let's replace the magic numbers with constants, defined at the top of the file. This will make the code easier to understand and maintain.